baseplate/clients/kafka.py

```diff
@@ -1,27 +1,37 @@
 import logging
 import time
 
+from concurrent.futures import wait
 from typing import Any
 from typing import Dict
+from typing import List
+from typing import Optional
 
 import confluent_kafka
 
 from confluent_kafka.avro import AvroProducer
 from confluent_kafka.avro import Producer
+from gevent.threadpool import ThreadPoolExecutor
 
 from baseplate import Span
 from baseplate.clients import ContextFactory
 from baseplate.lib import config
 
-
 logger = logging.getLogger(__name__)
 
 TOPIC_SUFFIX = "_avro"
 CHECK_INTERVAL_SECONDS = 0.001
 
 
 class AvroProducerContextFactory(ContextFactory):
-    def __init__(self, bootstrap_servers: str, schema_registry: str, acks: int):
+    def __init__(
+        self,
+        bootstrap_servers: str,
+        schema_registry: str,
+        acks: int,
+        thread_pool_max_size: int = 10,
+        prefetch_schema_ids: Optional[List[int]] = None,
```

Comment:
Nit: maybe type this as `Sequence` rather than `List`?  Then you can pass in a list or a tuple, just anything that you can iterate over.